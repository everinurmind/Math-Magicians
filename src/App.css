.App {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 600px;
  gap: 50px;
  background-color: #333;
  padding: 50px;
}
