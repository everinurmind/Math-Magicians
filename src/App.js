import './App.css';
import React from 'react';
import { Routes, Route } from 'react-router-dom';
import Header from './components/Header';
import Home from './components/Home';
import Calculator from './components/Calculator';
import Quote from './components/FetchApi';

function App() {
  return (
    <div className="App">
      <Header />
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/calculator" element={<Calculator />} />
        <Route path="/quote" element={<Quote />} />
      </Routes>
    </div>
  );
}

export default App;
