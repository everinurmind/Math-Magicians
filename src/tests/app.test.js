import renderer from 'react-test-renderer';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import App from '../App';

test('should render App.js', () => {
  const text = renderer.create(
    <BrowserRouter>
      <App />
    </BrowserRouter>,
  );

  expect(text.toJSON()).toMatchSnapshot();
});
test('should render Home component', () => {
  const text = renderer.create(
    <BrowserRouter>
      <Routes>
        <Route path="*" element={<App />} />
      </Routes>
    </BrowserRouter>,
  );

  expect(text.toJSON()).toMatchSnapshot();
});
