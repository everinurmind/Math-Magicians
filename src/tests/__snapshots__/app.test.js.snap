// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App.js 1`] = `
<div
  className="App"
>
  <header
    className="site-header"
  >
    <h1
      className="site-title"
    >
      Math Magicians
    </h1>
    <nav
      className="main-navigation"
    >
      <ul
        className="menu"
      >
        <li>
          <a
            aria-current="page"
            className="menu-item active"
            exact={true}
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
         | 
        <li>
          <a
            className="menu-item"
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
         | 
        <li>
          <a
            className="menu-item"
            href="/quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <div
    className="site-home"
  >
    <h2
      id="site-header"
    >
      Welcome to our page!
    </h2>
    <section
      className="text-section"
    >
      <p>
        Welcome to our React app! Our app features a navigation menu with three main pages: Home, Calculator, and Quotes.
      </p>
      <p>
        The Calculator page is a simple yet powerful calculator. Whether you need to add, subtract, multiply, or divide, our calculator has got you covered.
      </p>
      <p>
        The Quotes page is perfect for when you need a little bit of inspiration or motivation. With just a click of a button, you can generate a random quote.
      </p>
    </section>
  </div>
</div>
`;

exports[`should render Home component 1`] = `
<div
  className="App"
>
  <header
    className="site-header"
  >
    <h1
      className="site-title"
    >
      Math Magicians
    </h1>
    <nav
      className="main-navigation"
    >
      <ul
        className="menu"
      >
        <li>
          <a
            aria-current="page"
            className="menu-item active"
            exact={true}
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
         | 
        <li>
          <a
            className="menu-item"
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
         | 
        <li>
          <a
            className="menu-item"
            href="/quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <div
    className="site-home"
  >
    <h2
      id="site-header"
    >
      Welcome to our page!
    </h2>
    <section
      className="text-section"
    >
      <p>
        Welcome to our React app! Our app features a navigation menu with three main pages: Home, Calculator, and Quotes.
      </p>
      <p>
        The Calculator page is a simple yet powerful calculator. Whether you need to add, subtract, multiply, or divide, our calculator has got you covered.
      </p>
      <p>
        The Quotes page is perfect for when you need a little bit of inspiration or motivation. With just a click of a button, you can generate a random quote.
      </p>
    </section>
  </div>
</div>
`;
